src/array.cr

```diff
@@ -590,6 +590,51 @@ class Array(T)
     elem
   end
 
+  # Deletes all elements that are within the given range,
+  # returning that elements.
+  # Raises `IndexError` if the index is out of range.
+  #
+  # ```
+  # a = ["ant", "bat", "cat", "dog"]
+  # a.delete_at(1..2)  #=> ["bat", "cat"]
+  # a                  #=> ["ant", "dog"]
+  # a.delete_at(99..100)  #=> IndexError
+  # ```
+  def delete_at(range : Range)
+    return nil if range.first < 0 && range.end > 0
+    from = range.begin
+    from += size if from < 0
+    to = range.end
+    to += size if to < 0
+    to -= 1 if range.excludes_end?
+    len = to - from + 1
+    len = 0 if len < 0
+    delete_at(from, len)
+  end
+
+  # Deletes count or less (if there aren't enough) elements at the given start index, 
+  # returning that elements.
+  # Raises `IndexError` if the index is out of range.
+  #
+  # ```
+  # a = ["ant", "bat", "cat", "dog"]
+  # a.delete_at(1, 2)  #=> ["bat", "cat"]
+  # a                  #=> ["ant", "dog"]
+  # a.delete_at(99,1)  #=> IndexError
+  # ```
+  def delete_at(index, count)
+    val = self[index, count]
+    index = size + index if index < 0
+    return nil if index < 0
```

Comment:
I think I miss a spec for this case :) (indirectly via the range overload doesn't count ;) )